gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -267,4 +209,127 @@ private Instant getCutoffTimeForRef(String reference, Map<String, Instant> dropp
                 GCUtil.deserializeReference(reference).getName(),
                 gcParams.getDefaultCutOffTimestamp());
   }
+
+  // --- Methods for computing expired content rows ----------
+  /**
```

Comment:
> Mixtrure of Java and Scala iterators

I might have to write `DistributedIdentifyContents` and `IdentifyContentsPerExecutor` in scala :(

> Check and fix usage and lifetime of NessieApi

Yes, In driver I can close the NessieApi before calling `distributedIdentifyContents.identifyExpiredContents` in GCImpl.
Executor creates is creating NessieApi per reference, I can make it per task. 

>Ensure that only one Row for each content/snapshot-ID is returned

Currently rows are returned per reference, say if the snapshot id 42 is expired for contentId_1 in 5 different references, there will be 5 similar rows for contentID-snapshotID (contentId_1 - 42) with different ref@hash in the total output.

I can add distinct here, but we lose reference info if I do that, also adding aggregate function require computation of whole data. So, it will bloat up.
In the consumer (while querying the output table), we can easily get the distinct values.

Reply:
>Ensure that only one Row for each content/snapshot-ID is returned
Maybe you need to remodel the structure to capture the fact that you have a list of smth per the `content/snapshot-ID`?

Reply:
> Maybe you need to remodel the structure to capture the fact that you have a list of smth per the content/snapshot-ID?

That needs data shuffle (aggregate function) and needs entire data. It will bloat up. Hence, handing in the query is better as it will have filter + aggregate.